**DO NOT MERGE** [website] Alternative Developer Advocate job posting (#39210)
mui/material-ui: docs/pages/careers/developer-advocate.md

> # Developer Advocate

Author: I opted to keep the title open-ended, but we could potentially change this to something like `Content Developer` (for the content path) and/or `DX Engineer` (for the docs infra path)

We also host Design Kits and pre-built Templates.

Beyond the core libraries, MUI X offers advanced components like the Data Grid, Date and Time Pickers, and Charts, for more complex user interactions and data visualization needs.

We're also making ambitious moves to incorporate our full suite of components into Toolpad, a low-code admin builder tool for assembling full-stack apps faster than ever.

Learn more about MUI's products in this blog post: [An introduction to the MUI ecosystem](https://mui.com/blog/mui-product-comparison/). (This is also the kind of content you might create!)

## The culture

MUI is a fully remote company with a team that spans the globe.
The majority of our work is asynchronous, and we rely on written communication to collaborate.
We're radically transparent: nearly all of our work happens in public.
Each contributor has the freedom to decide how and when they work, and that work is primarily self-directed: it's your responsibility to define and complete your own tasks in a timely manner.

Learn more about what we value as an organization: [Evolving MUI's core values and behaviors](https://mui.com/blog/2023-mui-values/).

> ## Why we're hiring

Author: @oliviertassinari I think this sums up what we discussed about giving the applicant more meaningful context that they wouldn't want to just skip over. 😅

MUI's DevEx program is currently maintained by a "team" of one, and that one person devotes the majority of their time to documentation and written technical content.

> Our users consistently rank documentation as one of the most important factors in deciding whether or not to build with our products, and many users express dissatisfaction with the rather steep learning curve that they face when onboarding.

Reviewer: Maybe a line break, it feels like we are aiming to convey a different point there:

```suggestion

Our users consistently rank documentation as one of the most important factors in deciding whether or not to build with our products, and many users express dissatisfaction with the rather steep learning curve that they face when onboarding.
```

> With that in mind, we're looking to add a second individual to the DevEx team who can dig deep into the documentation and begin expanding our output of educational materials, to ease the learning curve and inspire users with great examples of what's possible with our component libraries.

Reviewer: More than beginning?

```suggestion
With that in mind, we're looking to add a second individual to the DevEx team who can dig deep into the documentation and expand our output of educational materials, to ease the learning curve and inspire users with great examples of what's possible with our component libraries.
```

Depending on your skills and interests, you may choose to pursue one of two paths on the DevEx team: content (creating educational resources) or infrastructure (improving the DX of the product docs and marketing site).

> (We expect to hire specialists for both of these paths eventually, so we're open to hiring for either one right now.)

Author: This is something @oliviertassinari and I have been talking about, but we can remove this part (or edit it to specify one of these two paths) if we don't want to proceed with this idea

MUI's Developer Experience program consists of three key pillars—Education, Documentation, and Community—which all contribute to the company's primary goal: to help developers build beautiful user interfaces quickly and efficiently.

As our next Developer Advocate, you will focus your efforts on education and documentation initiatives, though you may also work on projects related to product DX and community relations from time to time as well.

You will also have frequent opportunities to shape the DevEx roadmap and contribute to product roadmaps by gathering feedback from the community and voicing the needs of users.

> 

Reviewer: Could this make sense?

```suggestion

Refer to the Radiating Circles of DX model https://dx.tips/circles to better understand how we think about this role. You will help strengthen the Product DX and Docs layers and pioneer the Content layer.

```

### Key responsibilities

- Create educational materials (tutorials, how-to guides, demo apps) for devs of all skill levels
- Seek out and implement improvements to the documentation

> - Contribute to the docs infrastructure and UX

Reviewer: I'm pushing for a dedicated role, we could move this lower.

- Engage with the community and gather feedback for product teams
- Serve as the "voice" of the company through social media, conferences, and events

> 

Reviewer: ```suggestion

For the right candidate:

- Lead the Developer Experience team. Working with the Leadership to construct and execute a hiring plan to grow the team.

```

### Tasks and initiatives you might take on

- Create tutorials for migrating from competitor products
- Improve the Algolia search experience

> - Be a first responder for GitHub issues and PRs related to docs

Reviewer: Broader?

```suggestion
- Own community GitHub issues or PRs related to developer experience
```

- Build example apps with MUI products and create how-to content around them
- Research and write case studies
- Overhaul the [Showcase](https://mui.com/material-ui/discover-more/showcase/)
- Revamp the company blog infrastructure to empower less technical teammates to contribute

> - Solicit (and potentially contribute to) integrations with other popular libraries and frameworks

Reviewer: Considering we want the person to be able to go as deep as fixing the product, I think we can keep it broad "contribute", in any way that make sense, mostly either doing if nobody else as the bandwidth or only connecting the dots if the person time is better spent somewhere else

```suggestion
- Contribute to integrations with other popular libraries and frameworks
```

> 

Reviewer: This is well aligned no?

```suggestion
- Create a learning section in the documentation for hybrid written and video tutorials.

```

## Who we're looking for

> You're a React developer with a few years of professional experience on the front end, and maybe you've worked with design systems before.

Reviewer: This feels a bit duplicate with what comes next

Author: agreed, feels redundant - removed

You love teaching technical concepts to others, and you're patient and empathetic when working with devs of all skill levels.
You have a strong track record of published technical content (which might just be your own personal blog or YouTube channel where you share things you've learned).
You're a self-starter who's comfortable working independently and owning projects from end to end.

### Required

> 

Reviewer: ```suggestion

- You're passionate about UI development.
- You are excited at the idea of helping engineers build UIs that feel amazing to use.
```

- Fluent written and spoken English skills—native speakers preferred

> - Experience as a React developer

Reviewer: I think it's more than experience we need, but a good level.

```suggestion
- Hands-on developer who is a skilled React engineer.
```

- Experience as a teacher, tutor, or mentor
- Experience as a technical writer or content creator

> 

Reviewer: ```suggestion
- You are a highly empathic person.

```

### Nice to have (but not required)

- Great taste in product design
- Knowledge of design trends
- Experience in open source
- Experience with design systems
- Experience with MUI products
- Experience with public speaking
- Experience creating video content
- Experience in community building (online or in person)

## Benefits and compensation

We offer competitive compensation depending on the profile and location.
We're ready to pay top market rates for a person who can clearly exceed the role's expectations.

> Learn more about perks and benefits on the [careers](https://mui.com/careers/#perks-amp-benefits) page.

Reviewer: I changed this

```suggestion
Learn more about perks and benefits on the [careers](https://mui.com/careers/#perks-and-benefits) page.
```
